lightning/src/ln/channelmanager.rs

```diff
@@ -2679,7 +2718,7 @@ impl<ChanSigner: ChannelKeys, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref>
 						return Err(MsgHandleErrInternal::send_err_msg_no_close("Got a message for a channel from the wrong node!".to_owned(), msg.channel_id));
 					}
 					let was_frozen_for_monitor = chan.get().is_awaiting_monitor_update();
-					let (commitment_update, pending_forwards, pending_failures, closing_signed, monitor_update) =
+					let (commitment_update, pending_forwards, pending_failures, closing_signed, monitor_update, htlcs_to_fail) =
```

Comment:
Hmm, isn't it an issue that we may return early two lines down and thus not fail back the HTLCs which were returned?

Reply:
Hm yeah, that should be updated. Is that an existing problem for the pending failures [here](https://github.com/rust-bitcoin/rust-lightning/blob/master/lightning/src/ln/channelmanager.rs#L2709) too?